src/parachain/amm.ts

```diff
@@ -0,0 +1,443 @@
+    ): Promise<Array<MonetaryAmount<CurrencyExt>>> {
+        const pooledMonetaryCurrencies = await Promise.all(
+            currencyIds.map((currencyId) =>
+                currencyIdToMonetaryCurrency(this.assetRegistryAPI, this.loansAPI, currencyId)
+            )
+        );
+
+        const pooledCurrencies = pooledMonetaryCurrencies.map((currency, index) =>
+            newMonetaryAmount(Big(balances[index].toString()), currency)
+        );
+
+        return pooledCurrencies;
+    }
+
+    private async _getStablePoolAmplificationCoefficient(
+        lpTokenCurrencyId: InterbtcPrimitivesCurrencyId
+    ): Promise<Big> {
+        // TODO: Use rpc call rpc.zenlinkStableAmm.getA
+        throw new Error("Method not implemented.");
+    }
+
+    private async _getStableLiquidityPool(
+        poolId: number,
+        poolData: ZenlinkStableAmmPrimitivesPool
+    ): Promise<StableLiquidityPool | null> {
+        const poolBase = this._getStableBasePool(poolData);
+        if (poolBase === null) {
+            return null;
+        }
+
+        const [pooledCurrencyIds, pooledCurrencyBalances, tradingFee, lpTokenCurrencyId] = [
+            poolBase.currencyIds,
+            poolBase.balances,
+            // TODO: check number base for fee (is there a need to divide?)
+            Big(poolBase.fee.toString()),
+            poolBase.lpCurrencyId,
+        ];
+
+        const lpToken = this._getStableLPTokenFromPoolData(poolId, poolBase);
+
+        const [pooledCurrencies, apr, A, totalSupply] = await Promise.all([
+            this._getStablePoolPooledCurrencies(pooledCurrencyIds, pooledCurrencyBalances),
+            this._getStablePoolAPR(poolId),
+            this._getStablePoolAmplificationCoefficient(lpTokenCurrencyId),
+            this.tokensAPI.total(lpToken),
+        ]);
+
+        return new StableLiquidityPool(lpToken, pooledCurrencies, apr, tradingFee, poolId, A, totalSupply);
+    }
+
+    public async getStableLiquidityPools(): Promise<Array<StableLiquidityPool>> {
+        const poolEntries = await this.api.query.zenlinkStableAmm.pools.entries();
+        const rawPoolsData = poolEntries.filter(([_, pool]) => pool.isSome);
+        const pools = await Promise.all(
+            rawPoolsData.map(([poolId, poolData]) =>
+                this._getStableLiquidityPool(poolId.args[0].toNumber(), poolData.unwrap())
+            )
+        );
+
+        return pools.filter((pool) => pool !== null) as Array<StableLiquidityPool>;
+    }
+
+    async getLiquidityPools(): Promise<Array<LiquidityPool>> {
+        const [standardPools, stablePools] = await Promise.all([
+            this.getStandardLiquidityPools(),
+            this.getStableLiquidityPools(),
+        ]);
+        return [...standardPools, ...stablePools];
+    }
+
+    async swap(
+        trade: Trade,
+        minimumAmountOut: MonetaryAmount<CurrencyExt>,
+        recipient: AddressOrPair,
+        deadline: number | string
+    ): Promise<void> {
+        //TODO
```

Comment:
Would throwing not implemented make sense for this method (and the next two) make sense?
If so, that would make it consistent with the other unimplemented ones.